### ISSUE TITLE
[clang] False positive -Wdocumentation + tparam + explicit instantiation declaration

### ISSUE THREAD

**Issue by carlosgalvezp** [llvm/llvm-project#52977]
The `-Wdocumentation` warning gives a false positive when documenting a template parameter via `tparam`, and an explicit instantiation declaration is used. The warning goes away if I remove the explicit instantiation declaration.

[Minimal example](https://godbolt.org/z/47zanxGz6):

```cpp
// foo.h

/// Foo.
///
/// \tparam x Some value.
template <int x>
void foo();

// Explicit instantiation declaration.
extern template void foo<1>();

// foo.cpp

// #include "foo.h"
// Implementation
template <int x>
void foo()
{}

// Explicit instantiation
template void foo<1>();

```

Compiling:
```
$ clang++ -Wdocumentation -c foo.cpp
foo.cpp:5:6: warning: '\tparam' command used in a comment that is not attached to a template declaration [-Wdocumentation]
/// \tparam x Some value.
     ^~~~~~
```
